features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/APIDefinition/DefinitionOutdated.jsx

```diff
@@ -52,46 +53,87 @@ export default function DefinitionOutdated(props) {
     const [showDiff, setShowDiff] = useState(false);
     const [newDefinition, setNewDefinition] = useState('');
     const [oldDefinition, setOldDefinition] = useState('');
+    const [isRendering, setIsRendering] = useState(false);
     const { updateAPI } = useContext(APIContext);
+
+    /**
+     * Re import service definition
+     */
     function reimportService() {
         const promisedReimportService = API.reimportService(api.id);
         promisedReimportService.then(() => {
-            Alert.info('Re-import service successfully!');
+            Alert.info(
+                <FormattedMessage
+                    id='Apis.Details.APIDefinition.DefinitionOutdated.import.success'
+                    defaultMessage='Re-imported API definition successfully!'
+                />,
+            );
             setOpenImport(false);
         }).catch((error) => {
             if (error.response) {
                 Alert.error(error.response.body.description);
             } else {
-                const message = 'Error while Re-import service';
-                Alert.error(message);
+                Alert.error(
+                    <FormattedMessage
+                        id='Apis.Details.APIDefinition.DefinitionOutdated.import.error'
+                        defaultMessage='Error while re-importing API definition'
+                    />,
+                );
             }
-            console.error(error);
         }).finally(() => {
             updateAPI();
         });
     }
 
+    /**
+     * Show diff between old service definition and new service definition
+     */
     function showdiff() {
-        setShowDiff(true);
+        setIsRendering(true);
         const promisedServices = ServiceCatalog.searchServiceByKey(api.serviceInfo.key);
         promisedServices.then((data) => {
             return ServiceCatalog.getServiceDefinition(data.body.list[0].id).then((file) => {
-                setNewDefinition(JSON.stringify(file, null, 2));
+                api.getSwagger(api.id).then((resp) => {
```

Comment:
I think we can parallelize 

- `api.getSwagger(api.id)` and
- `ServiceCatalog.searchServiceByKey(api.serviceInfo.key)` 

requests ?? 
Looks like they are mutually exclusive ?

Let's use `Promise.all()` and make parallel calls it will improve the performance , reduce the wait time

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/10220/commits/e0ba14ceb09cefa92b3609a079d9cf333681d857